adcc/ExcitedStates.py

```diff
@@ -193,17 +208,75 @@ def __init__(self, data, method=None, property_method=None):
 
         # Special stuff for special solvers
         if isinstance(data, EigenSolverStateBase):
-            self.excitation_vectors = data.eigenvectors
-            self.excitation_energies = data.eigenvalues
+            self.__excitation_vector = data.eigenvectors
+            self.__excitation_energy = data.eigenvalues
         else:
             if hasattr(data, "eigenvalues"):
-                self.excitation_energies = data.eigenvalues
+                self.__excitation_energy = data.eigenvalues
             if hasattr(data, "eigenvectors"):
-                self.excitation_vectors = data.eigenvectors
-            if hasattr(data, "excitation_energies"):
-                self.excitation_energies = data.excitation_energies
-            if hasattr(data, "excitation_vectors"):
-                self.excitation_vectors = data.excitation_vectors
+                self.__excitation_vector = data.eigenvectors
+            if hasattr(data, "excitation_energy"):
+                # TODO: what if data has energy corrections?
+                # make better ctor if data is other ExcitedStates object
+                self.__excitation_energy = data.excitation_energy.copy()
+            if hasattr(data, "excitation_vector"):
+                self.__excitation_vector = data.excitation_vector
+
+        # Collect all excitation energy corrections
+        self.__excitation_energy_uncorrected = self.excitation_energy.copy()
+
+        if hasattr(self.reference_state, "excitation_energy_corrections"):
+            merged = self.reference_state.excitation_energy_corrections.copy()
+            merged.update(excitation_energy_corrections)
+            excitation_energy_corrections = merged
+
+        for key in excitation_energy_corrections:
+            corr_function = excitation_energy_corrections[key]
+            if not callable(corr_function):
+                raise TypeError("Elements in excitation_energy_corrections "
+                                "must be callable.")
+            # call the function for exc. energy correction
+            correction = np.array([
+                corr_function(exci) for exci in self.excitations])
+            setattr(self, key, correction)
+            self.__excitation_energy += correction
+
+        # FIXME: poor man's deprecation warnings
```

Comment:
I would do it no different, however I would put it below the class and not into the constructor.

Reply:
Done.